src/main/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -757,7 +757,7 @@ public static int hasBuildingEnoughElseCount(@NotNull final IBuilding provider,
 
                 if (totalCount >= count)
                 {
-                    return Integer.MAX_VALUE;
+                    return totalCount;
```

Comment:
I recall some callee's are using that max value return for logic

Reply:
I'll double check

Reply:
save